Feature/improve language localization for FR 20250607

jlebutte opened the pull request:
This pull request updates the French translations in the `apps/client/src/locales/messages.fr.xlf` file. The changes involve translating multiple UI strings from English to French, improving the localization of the application.

### Localization Updates:
* Translated "Demo user account has been synced." to "Le compte utilisateur de démonstration a été synchronisé."
* Translated "Sync Demo User Account" to "Synchroniser le compte utilisateur de démonstration."
* Translated "Emergency Fund: Set up" to "Fonds d'urgence : Configuration."
* Translated "No emergency fund has been set up" to "Aucun fonds d'urgence n'a été mis en place."
* Translated "The fees do not exceed ${thresholdMax}% of your initial investment (${feeRatio}%)" to "Les frais ne dépassent pas ${thresholdMax}% de votre investissement initial (${feeRatio}%)."

#3589

dtslvr commented on apps/client/src/locales/messages.fr.xlf:
```diff
@@ -7602,71 +7602,71 @@
       <trans-unit id="rule.emergencyFundSetup" datatype="html">
         <source>Emergency Fund: Set up</source>
-        <target state="new">Emergency Fund: Set up</target>
+        <target state="translated">Fonds d'urgence : Mise en place</target>
```
Please use `’` instead of `'`

dtslvr commented on apps/client/src/locales/messages.fr.xlf:
```diff
@@ -7602,71 +7602,71 @@
       <trans-unit id="rule.emergencyFundSetup.false" datatype="html">
         <source> No emergency fund has been set up </source>
-        <target state="new"> No emergency fund has been set up </target>
+        <target state="translated"> Aucun fonds d'urgence n'a été mis en place </target>
```
Please use `’` instead of `'`

dtslvr commented on apps/client/src/locales/messages.fr.xlf:
```diff
@@ -7602,71 +7602,71 @@
       <trans-unit id="rule.emergencyFundSetup.true" datatype="html">
         <source> An emergency fund has been set up </source>
-        <target state="new"> An emergency fund has been set up </target>
+        <target state="translated"> Un fonds d'urgence a été mis en place </target>
```
Please use `’` instead of `'`

dtslvr reviewed (changes requested):
Thanks for your contribution, @jlebutte! 👍🏻
I have reviewed your changes and added a few comments.

Could you please rebase the branch and move your changelog entry to the end of the `Changed` section?

jlebutte commented:
> Thanks for your contribution, @jlebutte! 👍🏻 I have reviewed your changes and added a few comments.
> 
> Could you please rebase the branch and move your changelog entry to the end of the `Changed` section?

Updated :) Thanks for the review

dtslvr reviewed (approved):
Thanks again for this contribution, @jlebutte! 🙌🏻
